java/trace/src/main/scala/org/typelevel/otel4s/java/ContextConversions.scala

```diff
@@ -24,33 +24,27 @@ import org.typelevel.vault.Vault
 
 private[otel4s] object ContextConversions {
 
-  def toJContext(context: Vault): JContext = {
-    var jContext = JContext.root()
-    jContext = Scope.fromContext(context) match {
-      case Scope.Root(_) =>
-        jContext
-      case Scope.Span(_, jSpan) =>
-        jSpan.storeInContext(jContext)
+  def toJContext(context: Vault): JContext =
+    Scope.fromContext(context) match {
+      case Scope.Root(ctx) =>
+        ctx
+      case Scope.Span(ctx, jSpan) =>
+        jSpan.storeInContext(ctx)
       case Scope.Noop =>
-        JSpan.getInvalid.storeInContext(jContext)
+        JSpan.getInvalid.storeInContext(JContext.root())
     }
-    jContext
-  }
 
-  def fromJContext(jContext: JContext): Vault = {
-    var context = Vault.empty
+  def fromJContext(jContext: JContext): Vault =
     JSpan.fromContextOrNull(jContext) match {
       case null =>
-        context = Scope.root.storeInContext(context)
+        Scope.Root(jContext).storeInContext(Vault.empty)
```

Comment:
the name `Root` becomes rather dubious at this point. like, this looks correct, but I don't think it's necessarily actually the root scope at this point

Reply:
It's still a root scope because there is no active span. But the root scope has some extra information (e.g. headers).

I agree the name `Root` getting misleading though.

Reply:
ah, good point. it's a root scope, but not necessarily a root context 😅